src/js/binaryen.js-post.js

```diff
@@ -2505,36 +2524,48 @@ function wrapModule(module, self = {}) {
   };
   self['getMemoryInfo'] = function() {
     var memoryInfo = {
-      'module': UTF8ToString(Module['_BinaryenMemoryImportGetModule'](module)),
-      'base': UTF8ToString(Module['_BinaryenMemoryImportGetBase'](module)),
-      'initial': Module['_BinaryenMemoryGetInitial'](module),
-      'shared': Boolean(Module['_BinaryenMemoryIsShared'](module))
+      'module': self['memory']['module'](),
+      'base': self['memory']['base'](),
+      'initial': self['memory']['initial'](),
+      'shared': self['memory']['shared']()
     };
-    if (Module['_BinaryenMemoryHasMax'](module)) {
-      memoryInfo['max'] = Module['_BinaryenMemoryGetMax'](module);
+    var max = self['memory']['max']();
+    if (max) {
```

Comment:
What is this checking? If the max is 0?

I'd expect this to check if the max is Infinity, but maybe I'm missing something...

Reply:
@kripken I'm not sure, I copied this from the existing code. Who wrote that?

Reply:
@kripken how about this?

```suggestion
    if (max !== Infinity) {
```

Reply:
FWIW, I guess the NFC value here would be `undefined`. Apart from this, if my other suggestions are agreed upon to be considered, there will be a `HasMax` function around that can be used instead, closely matching the previous code.